src/node.js

```diff
@@ -390,6 +391,78 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var unhandledPromises = new WeakMap();
+    // A hack to work around uniterable WeakMaps.
+    var unhandledPromiseKeys = [];
+    var fullTurnQueued = false;
+
+    function queueFullTurn() {
+      if (!fullTurnQueued) {
+        fullTurnQueued = true;
+        timers.setImmediate(function() {
+          process.nextTick(afterFullTurn);
+        });
+      }
+    }
+
+    function afterFullTurn() {
+      fullTurnQueued = false;
+      var keys = unhandledPromiseKeys;
+      unhandledPromiseKeys = [];
+      for (var i = 0; i < keys.length; ++i) {
+        var promise = keys[i];
+        var unhandledRejection = unhandledPromises.get(promise);
+        // A handler was added before a full turn happened.
+        if (unhandledRejection === undefined) continue;
+        unhandledRejection.hasBeenNotified = true;
+        var reason = unhandledRejection.reason;
+        if (!process.emit('unhandledRejection', reason, promise)) {
+          // Nobody is listening.
```

Comment:
It should throw. Unhandled promises are very similar to unhandled exceptions in this matter. 

There is only a question of backward compatibility, because people are already using promises. Maybe do `console.error("this will throw later")` for now and revisit this issue in the future versions.